python/private/pypi/evaluate_markers.bzl

```diff
@@ -0,0 +1,95 @@
+# Copyright 2024 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A simple function that evaluates markers using a python interpreter."""
+
+load("//python/private:repo_utils.bzl", "repo_utils")
+load(":pypi_repo_utils.bzl", "pypi_repo_utils")
+
+def evaluate_markers(ctx, *, requirements, python_interpreter, python_interpreter_target):
+    """Return the list of supported platforms per requirements line.
+
+    Args:
+        ctx: repository_ctx or module_ctx.
```

Comment:
Can we use a name other than `ctx`? That name is too strongly associated with analysis phase rules IMHO. `mrctx` ?